codegen_c.cpp unconditionally uses flag which may not be supported: `clang: error: unknown argument: '-fno-tree-sra'`

This breaks the build with Apple clang of 10.15:
https://github.com/thepowersgang/mrustc/blob/da4075f17328474ba90d2b3b56dde6be1954eeea/src/trans/codegen_c.cpp#L1297-L1301

```
--- BUILDING core v0.0.0 (0.0% 1r,0w,12b,0c/13t)
> /opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/bin/mrustc /opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/rustc-1.29.2-src/src/libcore/lib.rs -o output/libcore.rlib -C emit-depfile=output/libcore.rlib.d --cfg debug_assertions -O -L output --crate-name core --crate-type rlib --crate-tag 0_0_0 > output/libcore.rlib_dbg.txt
 (0.0% 1r,0w,12b,0c/13t): core v0.0.0
/opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/rustc-1.29.2-src/src/libcore/slice/mod.rs:1947-1948 warn:0:Unexpected attribute rustc_on_unimplemented on impl
/opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/rustc-1.29.2-src/src/libcore/slice/mod.rs:1960-1961 warn:0:Unexpected attribute rustc_on_unimplemented on impl
clang: error: unknown argument: '-fno-tree-sra'
C Compiler failed to execute - error code 256
Process exited with non-zero exit status 1
FAILING COMMAND: /opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/bin/mrustc /opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/rustc-1.29.2-src/src/libcore/lib.rs -o output/libcore.rlib -C emit-depfile=output/libcore.rlib.d --cfg debug_assertions -O -L output --crate-name core --crate-type rlib --crate-tag 0_0_0
Env:  OUT_DIR=/opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/output/build_core CARGO_MANIFEST_DIR=/opt/local/var/macports/build/mrustc-451cc08d/work/mrustc-da4075f17328474ba90d2b3b56dde6be1954eeea/rustc-1.29.2-src/src/libcore CARGO_PKG_NAME=core CARGO_PKG_VERSION=0.0.0 CARGO_PKG_VERSION_MAJOR=0 CARGO_PKG_VERSION_MINOR=0 CARGO_PKG_VERSION_PATCH=0
 (7.7% 0r,0w,12b,1c/13t):
BUILD FAILED
make: *** [LIBS] Error 1
```

--- a/src/trans/codegen_c.cpp
+++ b/src/trans/codegen_c.cpp
@@ -1314,11 +1314,15 @@ namespace {
                     args.push_back("-O1");  // HACK: Work around mrustc #347 by reducing the optimisation level
                     break;
                 }
+#if defined(__GNUC__) && !defined(__clang__)
+#if __GNUC__ < 16 && !(__GNUC__ == 15 && __GNUC_MINOR__ > 1)
                 // HACK: Work around [https://gcc.gnu.org/bugzilla/show_bug.cgi?id=117423] by disabling an optimisation stage
                 if( opt.opt_level > 0 )
                 {
                     args.push_back("-fno-tree-sra");
                 }
+#endif
+#endif
                 if( opt.emit_debug_info )
                 {
                     args.push_back("-g");

